### PR TITLE
Vasb 10255 add new poi functionality

### PR DESCRIPTION
**Pull request by JauntAR**
This PR addresses: http://jira.viewar.com/browse/VASB-10255

"Add New POI functionality"

- Click model to add POI
- Calculate furthest intersection
- Add separate "Add / Regular" modes to /MapEdit - but no extra routes
- Once a new POI is added, focus the new input field - using new hook "usePrevious"
- Default POI name is ""
- Added many checks for missing poi.data potential situations (unfortunately)

### CODE REVIEW

File: src/hooks/usePrevious.ts [hunk 1]
```diff
@@ -0,0 +1,9 @@
+import React, { useEffect, useRef } from "react";
```
Comment (kh-viewar) [inline src/hooks/usePrevious.ts:1]: We could've also used existing ones, e.g. https://www.npmjs.com/package/@hooks/previous

File: src/pages/MapEdit/MapEdit.tsx [hunk 2]
```diff
@@ -58,6 +62,10 @@ export const MapEdit: React.FC = () => {
 
   useEffect(() => {
     const initView = async () => {
+
+      // Shouldn't setCollisionMode() accept a boolean instead of number?
+      await viewarApi.appUtils.setCollisionMode(0); // Prevent models collision with scan
```
Comment (kh-viewar) [inline src/pages/MapEdit/MapEdit.tsx:67]: I would set this only once directly in index.tsx after viewarApi.init. It accepts a number it's actually using flags:
`Valid flags: 0 (COLLISION_DISABLED), 1 (COLLISION_ENABLED), 2 (COLLISION_NOTIFICATION)`
So you can combine 1 and 2 and pass in 3.

**Review by kh-viewar — APPROVED** [review]

**Comment by JauntAR** [comment]
I added one last commit where I did a preliminary refactor of the common / shared List items between POIs and Tours. This hopefully will save us time later when we need to update / modify the base EditListItem.tsx component

File: src/pages/MapEdit/MapEdit.tsx (hunk 2 above)
Comment (JauntAR) [reply src/pages/MapEdit/MapEdit.tsx:67]: Done

**Review by kh-viewar — APPROVED** [review]

**Pull request merged**